core/capabilities/remote/dispatcher.go

```diff
@@ -58,29 +60,80 @@ func (d *dispatcher) Start(ctx context.Context) error {
 		return fmt.Errorf("peer is not initialized")
 	}
 	d.wg.Add(1)
-	go d.receive()
+	go func() {
+		defer d.wg.Done()
+		d.receive()
+	}()
+
 	d.lggr.Info("dispatcher started")
 	return nil
 }
 
-func (d *dispatcher) SetReceiver(capabilityId string, donId string, receiver remotetypes.Receiver) error {
+func (d *dispatcher) Close() error {
+	close(d.stopCh)
+	d.wg.Wait()
+	d.lggr.Info("dispatcher closed")
+	return nil
+}
+
+var capReceiveChannelUsage = promauto.NewGaugeVec(prometheus.GaugeOpts{
+	Name: "capability_receive_channel_usage",
```

Comment:
Maybe prefix with "dispatcher_" instead of "capability_"?

Reply:
its the usage of the capabilities receive channel, so i this capability is the best name